src/Application/Job/CatalogDumpJob.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LinkedDataSets\Application\Job;
+
+use EasyRdf\Graph;
+use EasyRdf\RdfNamespace;
+use EasyRdf\Resource;
+use Laminas\Log\Logger;
+use Laminas\ServiceManager\Exception\ServiceNotFoundException;
+use Laminas\ServiceManager\ServiceLocatorInterface;
+use Omeka\Entity\Job;
+use Omeka\Job\AbstractJob;
+use Omeka\Job\Exception\InvalidArgumentException;
+
+final class CatalogDumpJob extends AbstractJob
+{
+    // in php 8.1 convert to enum
+    const DUMP_FORMATS = [
+        "turtle" => "ttl",
+        "ntriples" => "nt",
+        "jsonld" => "jsonld",
+        "rdfxml" => "xml",
+    ];
+
+    protected ?Logger $logger = null;
+    protected $serverUrl;
+    protected $id;
+
+    public function __construct(Job $job, ServiceLocatorInterface $serviceLocator)
+    {
+        parent::__construct($job, $serviceLocator);
+        $this->logger = $serviceLocator->get('Omeka\Logger');
+        if (!$this->logger) {
+            throw new ServiceNotFoundException('The logger service is not found');
+        }
+        $this->serverUrl = $serviceLocator->get('ViewHelperManager')?->get('ServerUrl');
+        if (!$this->serverUrl) {
+            throw new ServiceNotFoundException('The serverUrl service is not found');
+        }
+        $this->id = $this->getArg('id');
+        if (!$this->id) {
+            throw new InvalidArgumentException('No catalog_id was provided to the job');
```

Comment:
The exception message implies the required argument is named `catalog_id` where it actually is `id`. Consider renaming the argument or the exception message.